source/dub/generators/build.d

```diff
@@ -645,3 +650,40 @@ unittest { // issue #1235 - pass no library files to compiler command line when
 	auto gen = new BuildGenerator(prj);
 	gen.generate(settings);
 }
+
+version (Posix)
+private string signalToName (int signal) {
+	import core.stdc.signal;
+	import core.sys.posix.signal;
+
+	enum sig_types = [
```

Comment:
This function could be improved by look into what `core.stdc.signal` and `core.sys.posix.signal` exports instead of just hard-coding the list.

For example something this seems to work. But the codes layout / readability could be improved.
```d
string translate(int signal) {
    import core.sys.posix.signal;
    import std.algorithm : filter, sort;
    import std.array : array;
    import std.string : startsWith, indexOf;

    enum allSignals = [__traits(allMembers, core.sys.posix.signal), __traits(allMembers, core.stdc.signal)]
        .filter!(x => x.startsWith("SIG") && x.indexOf("_") < 0 )
        .array;

    switch (signal) {
        static foreach (sig; allSignals)
            static if (__traits(isIntegral, mixin(sig))) // SIGRTMIN / SIGRTMAX are functions on some systems, so ignore those.
                case mixin(sig):
                    return sig;
        default:
        	return "<Unknown signal>";
    }
}
```

Reply:
Hehe, a bit extreme. We can do something like this for now, but I hope we'll refactor those bindings in Druntime at some point. :)